### ISSUE TITLE
Db faker for functional tests

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#3591]
Many functional tests go through an elaborate process to get a workflow into the right state before doing various tests at or from that point.

Can we simplify this with a utility to fake the DB task pool table to any desired state, and take the scheduler there directly via restart?

**Comment by kinow** [comment]
Strange, I feel like I saw another issue somewhere about it? Or was it a thread on Riot? Anyway, +1 for this.

**Comment by oliver-sanders** [comment]
Note that this is possible at the moment but it requires generating many DB tables and faking lots of output. We could either:

* Enable Cylc to restart from a DB with fewer tables (e.g. just task pool),
* Write a wrapper which adds in the extra tables / fields as appropriate.

**Comment by hjoliver** [comment]
Without actually looking at the code, the `task_pool` table should be the only one that matters for a basic restart, so we should be able to get away with other tables existing but empty.

**Comment by kinow** [comment]
The other checkpoint tables are not used for restart too? (not sure if that's more than a basic restart)

**Comment by hjoliver** [comment]
The other tables are all(?) for stuff that might not exist, e.g. broadcasts are quite infrequently used, so we should be able to restart with those tables empty.

**Comment by hjoliver** [comment]
For initial investigations we can try messing with an existing DB for restart.  In sqlite, this statement completely empties a table:
```
DELETE FROM table_name;
```

**Comment by oliver-sanders** [comment]
> Without actually looking at the code, the task_pool table should be the only one that matters for a basic restart

Yep, though for some tests we may want to add other things like action timers.

**Comment by oliver-sanders** [comment]
I took a quick look at this whilst working on #3654 and discovered it's actually quite simple. Porting it to the bash framework in the functional tests might be a little more interesting, but still do-able.
